### DISCUSSION TITLE
Was hat es mit den LED auf sich?

### DISCUSSION THREAD

**Discussion by jblond4711** [tbnobody/OpenDTU#773] [General]
Hi,

kan. mir mal bitte jemand erklären was es mit den LED auf sich hat! Ich werde da nicht schlau draus. Muss ich zusätzliche LED anschließen an PIN 25 und 26 um etwas zu sehen?
Auf dem Display wird durch die neue Funktion aber nichts angezeigt oder doch?

**Comment by Juergen2453** [comment]
Die Doku

https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md
https://github.com/tbnobody/OpenDTU/blob/93c69dbbeb7bfea6ac552c28e9a779013455012b/docs/DeviceProfiles/blinkyparts_esp32.json

In der pin_mapping.json eintragen dort wo die LEDs dran sind.

**Comment by jblond4711** [comment]
Hi,
die Doku habe ich ja gelesen nur hapert es mit dem verstehen.
Muss ich extra LED‘s anschließen?

**Comment by Juergen2453** [comment]
Du kannst LEDs anschließen du musst es aber nicht.
Oder meinst du die LEDs auf dem ESP32.
Da gibt es verschiedene ESP32:
mit einer LED leuchtet immer (fest verdrahtet).
mit zwei LED eine leuchtet immer (fest verdrahtet) die andere kann man beeinflussen.

**Comment by jblond4711** [comment]
Hi,

ich wollte ja gern wissen ob ich für die LED Funktion, die jetzt eingebaut wurde, zusätzlich (optional) LED anschließen kann oder ob die LED auf dem ESP32 verwendet werden.
Vermutlich handelt es sich aber um optionale LED da ja die Pins 25 und 26 genutzt werden.

**Comment by Juergen2453** [comment]
Du kannst doch eintragen was du willst. Ich habe bei mir
led0 = 17
led1 = 18
eingetragen.

**Comment by huckes** [comment]
Hi, man kann die beiden LEDs auch als Warnanzeigen nutzen, dann leuchten/blinken sie nur im Fehlerfall bzw. beim Start des Moduls. Geht auch ohne Eingriff in den Sourcecode.
Einfach die LEDs (Polarität beachten) mit Vorwiderstand gegen +3,3V schalten.

**Comment by Juergen2453** [comment]
Bei openDTU kann man in der pin_mapping.json alles definieren ohne Eingriff in den Sourcecode.

**Comment by huckes** [comment]
Mit dem pin_mapping kann man die Pins definieren, klar. Aber nicht, wann die LEDs an sind. Ich denke, es kann Sinn machen, die LEDs nur im Fehlerfall leuchten zu lassen, insbesondere, wenn man ein Display verbaut hat. Dafür war der Hinweis gedacht, für diesen Fall die LEDs gegen +3,3V zu verdrahten.

**Comment by Juergen2453** [comment]
Das reicht doch als Fehlerfall.
https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md

**Comment by lethuer** [comment]
Ich habe auch eine Verständnisfrage...
Wie beschrieben habe ich die zwei externen LED an Pin25 + 26 meines ESP angeschlossen.
Also 3,3V zum Vorwiderstand, dann LED (Anode). LED (Kathode) dann an Pin 25 bzw. 26.

Zusätzlich in der pin_mapping das hier eingetragen und das Profil ausgewählt.:
```
    {
        "name": "Only LEDs",
        "nrf24": {
            "miso": 19,
            "mosi": 23,
            "clk": 18,
            "irq": 16,
            "en": 4,
            "cs": 5
        },
        "led": {
            "led0": 25,
            "led1": 26
        }
    },
```

Habe ich die LED richtig angeschlossen und die Kathodenseite wird auf - gezogen wenn die LED blinken soll oder ist an Pin25 bzw. 26 + und die Kathodenseite der LED muss an - ?

Bei mir sind so nun jedenfalls beide LED dauerhaft an, ich habe noch kein WLAN konfiguriert und es kann auch keine Verbindung zum WR bestehen.
Nach der Beschreibung sollten sie aber eigentlich beide aus sein ?!

**led.led0**	number	LED pin for network indication. Blinking = WLAN connected but NTP & MQTT (if enabled) disconnected. On = WLAN, NTP, MQTT connected. Off = Network not connected
**led.led1**	number	LED pin for inverter indication. On = All inverters reachable & producing. Blinking = All inverters reachable but not producing. Off = At least one inverter is not reachable. Only inverters with polling enabled are considered.

**Reply by micheld93** [reply]
Ich denke, die LEDs sind falsch angeschlossen. Das hier sollte die Verkabelung sein. Die 3,3V kommen dann über den GPIO über die LED zu GND und dann ist der Kreis zu. https://www.smarthome-tricks.de/esp8266/esp8266-blinkende-led/

**Reply by 3DJupp** [reply]
Wäre cool, wenn man die PIN config auch noch später ändern könnte, wie zum Beispiel bei Tasmota.

Könnte die LED/Board Verschaltung und Config (z.B in VS Code) nicht mit in die Doku mit aufgenommen werden?
💡

**Reply by tbnobody** [reply]
> Wäre cool, wenn man die PIN config auch noch später ändern könnte, wie zum Beispiel bei Tasmota.

Kann man ja... einfach im Device Profile die Pins setzen...
https://github.com/tbnobody/OpenDTU/blob/master/docs/DeviceProfiles.md

> Könnte die LED/Board Verschaltung und Config (z.B in VS Code) nicht mit in die Doku mit aufgenommen werden?

Klar. Schreibe eine Doku und ergänze sie hier: https://github.com/tbnobody/OpenDTU/wiki/LEDs

**Reply by micheld93** [reply]
@Dom1n1c Die Pin-Belegung kann wie von tbnobody beschrieben, geändert werden. Das hilft aber nicht, wenn der die Schaltung dann trotzdem nicht stimmt ;-).
